src/cljs/athens/views/athena.cljs

```diff
@@ -202,13 +200,37 @@
       (do (dispatch [:athena/toggle])
           (navigate-uid (or (:block/uid (:block/parent item)) (:block/uid item))))
 
-      ;; TODO: change scroll as user reaches top or bottom
-      ;; TODO: what happens when user goes to -1? or past end of list?
       (= key KeyCodes.UP)
-      (swap! state update :index dec)
+      (do
+        (.. e preventDefault)
+        (swap! state update :index #(dec (if (zero? %) (count results) %)))
+        (let [cur-index (:index @state)
+
+              ;; Search input box
+              input-el (.. e -target)
+
+              ;; Get the result list container which is the last element child
+              ;; of the whole athena component
```

Comment:
This makes a lot more sense to me now. I'm pretty sure I wrote `[:div.athena]`, but that should probably be an HTML id. `.athena` also doesn't do anything functionally, so maybe not that useful.

This could also be simplified if we just gave an HTML id/class to the results list directly, but not important.